src/Misc.pm

```diff
@@ -4353,14 +4353,28 @@ sub checkMonsterCondition {
 	my ($prefix, $monster) = @_;
 
 	if ($config{$prefix . "_hp"}) {
-		return 0 if (!$monster->{hp});
-		if ($config{$prefix . "_hp"} =~ /^(.*)\%$/) {
-			return 0 unless (inRange(($monster->{hp} * 100 / $monster->{hp_max}), $1)); 
-		} else {
-			return 0 unless (inRange($monster->{hp}, $config{$prefix . "_hp"})); 
-		}
+		eval {
+			if($config{$prefix . "_hp"} =~ /(\d+)%$/) {
+				if($monster->{hp} && $monster->{hp_max}) {
+					return 0 unless (inRange(($monster->{hp} * 100 / $monster->{hp_max}), $config{$prefix . "_hp"}));
+				} elsif($monster->{hp_percent}) {					
+					return 0 unless (inRange($monster->{hp_percent}, $config{$prefix . "_hp"}));
+				} else {
+					return 0;
+				}
+			} elsif($config{$prefix . "_hp"} =~ /(\d+)$/) {
+				if($monster->{hp} && $monster->{hp_max}) {
+					return 0 unless (inRange($monster->{hp}, $config{$prefix . "_hp"}));
+				} else {
+					return 0;
+				}
+			} else {
+				warning TF("%s is wrong configured.\n", $prefix . "_hp");
```

Comment:
We've never warned about invalid config blocks, I don't think we should start now, at least not in broken English